test/unit/SavingCirclesUnit.t.sol

```diff
@@ -526,4 +526,494 @@ contract SavingCirclesUnit is Test {
+    token.mint(alice, DEPOSIT_AMOUNT);
+    token.mint(bob, DEPOSIT_AMOUNT);
+    token.mint(carol, DEPOSIT_AMOUNT);
+
+    // Alice opts in and approves
+    vm.startPrank(alice);
+    savingCircles.setAutomatedDepositsEnabled(true);
+    token.approve(address(savingCircles), DEPOSIT_AMOUNT);
+    vm.stopPrank();
+
+    // Bob opts in but doesn't approve
+    vm.prank(bob);
+    savingCircles.setAutomatedDepositsEnabled(true);
+
+    // Carol approves but doesn't opt in
+    vm.prank(carol);
+    token.approve(address(savingCircles), DEPOSIT_AMOUNT);
+
+    // Get eligible addresses
+    (uint256[] memory circleIds, address[] memory eligibleMembers) = savingCircles.getEligibleAddressesForDeposit();
+
+    // Only alice should be eligible (opted in AND has allowance)
+    assertEq(eligibleMembers.length, 1);
+    assertEq(eligibleMembers[0], alice);
+    assertEq(circleIds[0], baseCircleId);
+  }
+
+  function test_GetEligibleAddressesForDeposit_MultipleCircles() external {
+    // Create second circle
+    address[] memory secondMembers = new address[](2);
+    secondMembers[0] = alice;
+    secondMembers[1] = bob;
+
+    ISavingCircles.Circle memory secondCircle = ISavingCircles.Circle({
+      owner: carol,
+      members: secondMembers,
+      currentIndex: 0,
+      circleStart: block.timestamp,
+      token: address(token),
+      depositAmount: DEPOSIT_AMOUNT / 2,
+      depositInterval: DEPOSIT_INTERVAL,
+      maxDeposits: MAX_DEPOSITS
+    });
+
+    vm.prank(carol);
+    uint256 secondCircleId = savingCircles.create(secondCircle);
+
+    // Setup approvals
+    token.mint(alice, DEPOSIT_AMOUNT * 2);
+    token.mint(bob, DEPOSIT_AMOUNT);
+
+    // Alice opts in and approves
+    vm.startPrank(alice);
+    savingCircles.setAutomatedDepositsEnabled(true);
+    token.approve(address(savingCircles), DEPOSIT_AMOUNT * 2);
+    vm.stopPrank();
+
+    // Bob opts in and approves
+    vm.startPrank(bob);
+    savingCircles.setAutomatedDepositsEnabled(true);
+    token.approve(address(savingCircles), DEPOSIT_AMOUNT / 2);
+    vm.stopPrank();
+
+    // Get eligible addresses
+    (uint256[] memory circleIds, address[] memory eligibleMembers) = savingCircles.getEligibleAddressesForDeposit();
+
+    // Should have 3 eligible entries: alice in both circles, bob in second circle
+    assertEq(eligibleMembers.length, 3);
+  }
+
+  function test_GetEligibleAddressesForDeposit_OutsideDepositWindow() external {
+    // Setup approvals
+    token.mint(alice, DEPOSIT_AMOUNT);
+
+    vm.startPrank(alice);
+    savingCircles.setAutomatedDepositsEnabled(true);
+    token.approve(address(savingCircles), DEPOSIT_AMOUNT);
+    vm.stopPrank();
+
+    // Move time past deposit window
+    vm.warp(block.timestamp + DEPOSIT_INTERVAL + 1);
+
+    // Get eligible addresses - should be empty
+    (uint256[] memory circleIds, address[] memory eligibleMembers) = savingCircles.getEligibleAddressesForDeposit();
+    assertEq(eligibleMembers.length, 0);
```

Comment:
also check "circleIds" length for good measure and to get rid of the warning abvoe